### DISCUSSION TITLE
It seems that otel.metrics.exporter=none is not working

### DISCUSSION THREAD

**Discussion by HyoTaek-Jang** [open-telemetry/opentelemetry-java-instrumentation#11888] [Q&A]
Hello,

I only need trace data and do not need to send metrics and logs data to OpenTelemetry (OTel). According to the documentation at https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection, setting the exporter to “none” should achieve this.

However, the following three methods did not work:

	1.	Using -Dotel.metrics.exporter=none
	2.	Setting the environment variable OTEL_METRICS_EXPORTER=none
	3.	Configuring otel.metrics.exporter=none in application.yaml

Despite these attempts, metrics logs are still being generated. How can I prevent metrics and logs from being sent to the OTel collector?

<img width="1655" alt="image" src="https://github.com/user-attachments/assets/0e0d6fd3-21fb-4624-8bd0-43092a4d3959">
<img width="1605" alt="image" src="https://github.com/user-attachments/assets/2aea2e3e-e274-4074-befd-a7cf0e494f06">

**Comment by laurit** [comment]
Are you using the otel java agent or something else?

**Reply by HyoTaek-Jang** [reply]
yes, I'm using the otel java agent. Is this a setting that doesn't work with java agent?

**Reply by laurit** [reply]
> yes, I'm using the otel java agent. Is this a setting that doesn't work with java agent?

It works with the java agent, but it does not if you manually configure otel sdk without using the autoconfigure extension.
Anyway the `http.server.request` metric that you have on the screenshot should be from micrometer not otel java agent.

**Reply by HyoTaek-Jang** [reply]
Thank you very much for your response.

Yes, the metrics in the screenshot are part of the overall metrics, and they are indeed coming from Micrometer.
I thought that setting OTEL_METRICS_EXPORTER=none would prevent any metrics from being exported to the OTel collector by the Java client. However, that was not the case.

So, how can I configure the Java client to send only traces and not send metrics and logs?
Do you think this issue is because it is configured with the OTel SDK?

**Reply by trask** [reply]
> they are indeed coming from Micrometer

hi @HyoTaek-Jang, I'd recommend asking the Micrometer folks about this (or you can use the OpenTelemetry Metrics SDK instead 😉)

**Reply by HyoTaek-Jang** [reply]
Okay, thank you for advise~!

**Reply by laurit** [reply]
I think you should google something like "spring boot disable metrics"

**Reply by HyoTaek-Jang** [reply]
Thank you all for your responses.

For those with a similar question, I’ll leave the solution here.
If you are using Micrometer, you also need to add the setting `-Dmanagement.otlp.metrics.export.enabled=false`. I initially thought the default was false, but it turns out it’s true.
